File: ex5.js
function PrimePalindrome(maxNumber) {
    this.maxNumber = maxNumber;
};

PrimePalindrome.prototype.isPalindrome = function (number) {
    array = number.toString().split('');
    return array.join('') == array.reverse().join('');
};

PrimePalindrome.prototype.isPrime = function (number) {
    for (var i = 2; i < number; i++) {
        if (number % i == 0)
            return false
    }
    return true
};

PrimePalindrome.prototype.isPrimePalindrome = function (number) {
  return this.isPrime(number) && this.isPalindrome(number);
};

PrimePalindrome.prototype.maxPrimePalindrome = function () {
  for (var i = this.maxNumber - 1; i < this.maxNumber; i--) {
      if (this.isPrimePalindrome(i))
        return i;
  }
};

console.log(new PrimePalindrome(999).maxPrimePalindrome());

File: ex5.rb
class PrimePalindrome
  attr_accessor :numbers

  def initialize(numbers)
    @numbers = numbers
  end

  def to_s
    numbers.to_a.reverse.each do |number|
      return number if is_prime_palindrome?(number)
    end
  end

  private

  def is_prime_palindrome?(number)
    is_palindrome?(number) && is_prime?(number)
  end

  def is_palindrome?(num)
    num.to_s == num.to_s.reverse
  end

  def is_prime?(number)
    arr = (2..number).select do |num|
      number % num == 0
    end
    arr.one?
  end
end

numbers = (1..1000)
puts PrimePalindrome.new(numbers).to_s
